include/seqan3/alignment/pairwise/detail/policy_alignment_matrix.hpp

```diff
@@ -149,5 +154,23 @@ class policy_alignment_matrix
 
         return std::tie(alignment_matrix, index_matrix);
     }
+
+    /*!\brief Checks whether the band is valid for the given sequence sizes.
+     *
+     * \param[in] sequence1_size The size of the first sequence.
+     * \param[in] sequence2_size The size of the second sequence.
+     *
+     * \throws seqan3::invalid_alignment_configuration if the band is invalid for the given sequence sizes and the
+     *         alignment configuration.
+     */
+    void check_valid_band_configuration(size_t const sequence1_size, size_t const sequence2_size) const
+    {
+        bool const upper_diagonal_ends_before_last_cell = (upper_diagonal + sequence2_size) < sequence1_size;
+        bool const lower_diagonal_ends_behind_last_cell = (-lower_diagonal + sequence1_size) < sequence2_size;
```

Comment:
Imagine having 1mio sequences and the last one is the odd-one-out, i.e. the band does not fit, and you get an exception that you can't handle gracefully (Even if you handle that exception, you can't resume the computation). This would be horrible! I would prefer to give an "infinite" score back and an empty alignment, instead of some exception. I suggested over and over again modelling the `infinite` score either as `std::optional` or as `the largest int score`, because of this and now you hack something that does not account for this and it will be a harder and harder to get this semantics into the code.

Reply:
As I said in the other PR. The plan after the features are there is to not throw at all but propagate the error state. This has the ultimate advantage, that we don't throw in multi-threaded library code but the user has the full control over it.